### PR TITLE
app/nextversion: make `next-version` error if there are no previous versions

### PR DESCRIPTION
**Pull request by roobre**
As discussed in #132.

Fun fact: I used https://github.com/zurawiki/gptcommit to try to write the commit message for me. Unfortunately that's not what you;'re seeing. The message gptcommit suggested was this ~extremely unrelated piece of crap~:

```
Improve recordings and completions accuracy

- Refactored the authentication code to use a more secure hashing algorithm
- Implemented a new API endpoint for user authentication
- Updated the database schema to store user authentication data securely

```

### CODE REVIEW

File: src/app/nextversion/nextversion.go [hunk 1]
```diff
@@ -129,6 +130,13 @@ func NextVersion(cCtx *cli.Context) error {
 	bmpr.DependencyCap = dependencyCap
 
 	next, err := bmpr.BumpSource(versionSrc)
```
Comment (roobre) [inline src/app/nextversion/nextversion.go:132]: I think we were previously ignoring this error. Some of the thousand linters should have caught this, right? Or am I missing something?

Comment (roobre) [reply src/app/nextversion/nextversion.go:132]: I was missing something, it is handled in the switch statement. I added a comment to clarify this.

**Review by alvarocabanas — CHANGES_REQUESTED** [review]
The changes LGTM, but I would add to the cli command description and to the general Readme and the action readme this behavior and/or the need to create a first version tag.

**Comment by roobre** [comment]
Good catch @alvarocabanas! Should be fixed in https://github.com/newrelic/release-toolkit/pull/133/commits/58161659954e7b9381b2fe5601c5b4af66c548b6

File: src/app/nextversion/nextversion_test.go [hunk 2]
```diff
@@ -318,10 +318,48 @@ dependencies:
...
+	}
+}
+
+//nolint:paralleltest // urfave/cli cannot be tested concurrently.
+func TestNextVersion_Fails(t *testing.T) {
+	for _, tc := range []struct {
+		name     string
+		yaml     string
+		args     string
+		expected string
```
Comment (sigilioso) [inline src/app/nextversion/nextversion_test.go:358]: Nit: we don't need a string expected output in this set of tests at the moment, do we?

File: src/app/nextversion/nextversion_test.go [hunk 3]
```diff
@@ -346,7 +383,7 @@ dependencies:
 		tc := tc
 		//nolint:paralleltest // urfave/cli cannot be tested concurrently.
 		t.Run(tc.name, func(t *testing.T) {
-			repoDir := repoWithTags(t, tc.tags...)
+			repoDir := t.TempDir()
```
Comment (sigilioso) [inline src/app/nextversion/nextversion_test.go:386]: I expected the repository to be initialized with something like

```go
repoDir := repoWithTags(t) // repo created with no tags
```

But now I reckon it to be a different test case.  As I see it, a repository with no tags and a directory (which is not a git repository) are different failure test cases. 

If we wanted to cover both, we could define the repository initialization as one of the test arguments so we could use:

```go
repoDir := tc.repo(t)
```

WDYT?

**Review by sigilioso — COMMENTED** [review]
Thanks for taking care of it 👍 
I've left a couple of comments related to the tests.

**Review by alvarocabanas — APPROVED** [review]
LGTM, once you tackle the comments, for me is good to go

File: src/app/nextversion/nextversion_test.go (hunk 2 above)
Comment (roobre) [reply src/app/nextversion/nextversion_test.go:358]: Good catch!

File: src/app/nextversion/nextversion_test.go (hunk 3 above)
Comment (roobre) [reply src/app/nextversion/nextversion_test.go:386]: I think it's a good point, I should have used a repo with no tags. I'm not sure why I went for the empty folder 🤔 

From what the command is concerned, there shouldn't be any difference though. But... it's easy enough to just provide the function :P

Comment (roobre) [reply src/app/nextversion/nextversion_test.go]: Done in https://github.com/newrelic/release-toolkit/pull/133/commits/8527223216e8d02eb0c2a17faa5d634e8be0cac3

**Review by sigilioso — APPROVED** [review]
🚀

**Pull request merged**